sdk/core/Azure.Core/src/Shared/OperationInternalBase.cs

```diff
@@ -146,8 +166,8 @@ public virtual async ValueTask<Response> WaitForCompletionResponseAsync(TimeSpan
                     return response;
                 }
 
-                TimeSpan delay = GetServerDelay(response, pollingInterval);
-                await WaitAsync(delay, cancellationToken).ConfigureAwait(false);
+                TimeSpan strategyInterval = pollingStrategy.GetNextWait(response);
```

Comment:
this might be better if we just instantiate a pollingstrat here since we need to call the choose logic to know if the passed in value will be used or ignored